axelarnetwork/axelar-core
# refact!: Unified Dockerfile, entrypoint for debug...

- Single Dockerfile for prod, debug and rosetta, This will enable single change instead of changing one thing at different files
  - Use single `RUN` step to create efficient layering
  - Use single command for `mkdir` and `chown` with multiple args instead of multiple single commands for each arg
  - Better caching at go mod, go build, .cache and apk level e.g. `make build` step without cache took `~32s` with cache `~7s`. This cache mechanism is `mount`, which mean for each change it'll mount cache instead of wiping cache on each change like some system does.
  - Use latest instead of experimental docker syntax
- Add ldflag `-s -w` and buildflag `-trimpath` to slim down binary size `~78.6M` -> `~58.0M`
- merge entrypoint file for debug and prod, since they share ~85% same code.
  - Enable it using `DEBUG_MODE=true`

!- Use alpine base for debug image

This change should make container image efficient, light without breaking anything but still need someone from internal team to verify especially with `entrypoint.sh`.

## Description

## Todos

- [ ] Upgrade handler

## Steps to Test

## Expected Behaviour

## Other Notes
